spec/order_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'order'
+
+describe Order do
+
+  let(:carbonara) { double('Carbonara', name: 'Carbonara', price: '10.50')}
+  let(:forcaccia) { double(name: 'Forcaccia', price: '2.00') }
+  let(:lasagne) { double(name: 'Lasagne', price: '8.90') }
+  let(:italian_menu) { double('Italian', items: [carbonara, forcaccia, lasagne]) }
+  let(:messenger) { double() }
+  let(:subject) { described_class.new(italian_menu, messenger) }
+  it { is_expected.to respond_to(:order_items) }
+  it { is_expected.to respond_to(:add_item).with(2).arguments }
+
+  it 'is bound to a menu on creation' do
+    subject = described_class.new(italian_menu)
+    expect(subject.menu).to eq(italian_menu)
+  end
+
+  describe '#add_item' do
+    it 'it adds 3 orders of "Carbonara" to order_items' do
+      subject = described_class.new(italian_menu)
+      expect { subject.add_item("Carbonara", 3) }.to change { subject.order_items }.to include(carbonara => 3)
+    end
+
+    it 'it adds 1 order of "Forcaccia" to order_items' do
+      subject = described_class.new(italian_menu)
+      expect { subject.add_item('Forcaccia', 1) }.to change { subject.order_items }.to include(forcaccia => 1)
+    end
+
+    it 'it adds 1 order of "Lasagne" and then another order' do
+      subject = described_class.new(italian_menu)
+      subject.add_item('Lasagne', 1)
+      expect { subject.add_item('Lasagne', 1) }.to change { subject.order_items }.to include(lasagne => 2)
+    end
+
+    it 'raises an error if an item not on the menu is requested' do
+      subject = described_class.new(italian_menu)
+      expect { subject.add_item('Gelato', 1) }.to raise_error('Item not on menu')
+    end
+  end
+
+  describe '#order_total' do
```

Comment:
A lot of setup with some of these tests, could maybe have utilised a before hook